app/models/pipeline_run.rb

```diff
@@ -121,9 +132,31 @@ class PipelineRun < ApplicationRecord
   STATUS_CHECKED = 'CHECKED'.freeze
   STATUS_FAILED = 'FAILED'.freeze
   STATUS_RUNNING = 'RUNNING'.freeze
-  STATUS_RUNNABLE = 'RUNNABLE'.freeze
+  STATUS_RUNNABLE = 'RUNNABLE'.freeze # TODO: (gdingle): not used anywhere?
   STATUS_READY = 'READY'.freeze
-
+  # NOTE: The current stored job_status are...
+  # +-------------------------------------------+
+  # | job_status                                |
+  # +-------------------------------------------+
+  # | 1.Host Filtering-FAILED                   |
+  # | 1.Host Filtering-FAILED|READY             |
+  # | 2.GSNAPL/RAPSEARCH alignment-FAILED       |
+  # | 2.GSNAPL/RAPSEARCH alignment-FAILED|READY |
+  # | 2.GSNAPL/RAPSEARCH alignment-RUNNING      |
+  # | 3.Post Processing-FAILED                  |
+  # | 3.Post Processing-FAILED|READY            |
+  # | 4.De-Novo Assembly-FAILED|READY           |
+  # | 4.Experimental-FAILED                     |
+  # | 4.Experimental-FAILED|READY               |
+  # | 4.Experimental-SUCCEEDED                  |
+  # | 4.Experimental-SUCCEEDED|READY            |
+  # | CHECKED                                   |
+  # | FAILED                                    |
+  # | LOADED                                    |
+  # +-------------------------------------------+
+  # NOTE: kickoff_pipeline does not set a job_status
+  validates :job_status, presence: true, allow_nil: true, if: :mass_validation_enabled?
```

Comment:
I'm a bit confused about this line, how can `presence: true` and `allow_nil: true` be in the same place?

Reply:
Ha, very good question! In doing this PR, I learned that `presence` will also catch whitespace and empty string. I think this is marginally beneficial in this case.

Reply:
Update: I'm working on PipelineMonitor right now and it seems that domain list for `job_status` field is incomplete. Just confirming, you are not validating that domain yet, correct?

Reply:
That's correct. I'm only validating `presence`. Those values are a snapshot for docs only.